agent/hcp/manager.go

```diff
@@ -71,15 +77,80 @@ func NewManager(cfg ManagerConfig) *Manager {
 	}
 }
 
+// runReporter initializes, and if successful, then runs the metrics reporter.
+func (m *Manager) runReporter(ctx context.Context) error {
+	// Step 1: Obtain CCM telemetry configuration
+	// Only enable HCP metrics reporting if server is registered with management plane.
+	telemetryCfg, err := m.cfg.Client.FetchTelemetryConfig(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to obtain CCM telemetry config: %v", err)
+	}
+
+	if telemetryCfg == nil {
+		// Not an error, as the server is not registered with CCM, but early return.
+		return nil
+	}
+
+	// Step 2: Init telemetry.MetricsExporter which sends metrics to HCP Metrics Gateway in OTLP format.
+	// It uses a an OTLP exporter client wrapped by the HCP client.
+	// This enables us to perform HCP auth and easily mock the client interface for tests.
+	// It must first be initialized within the HCP Client with the configured endpoint.
+	if err := m.cfg.Client.InitMetricsClient(ctx, telemetryCfg.Endpoint); err != nil {
+		return fmt.Errorf("failed to init metrics HCP client: %v", err)
+	}
+
+	expCfg := &telemetry.MetricsExporterConfig{
+		Labels: map[string]string{
+			"service.name":        "consul-server",
+			"service.version":     version.GetHumanVersion(),
+			"service.instance.id": m.cfg.NodeID,
```

Comment:
Labels are attributes that the metrics will be tagged with, for example the service instance id (`service.instance.id`)